### PR TITLE
Fixes brains counting towards highlander killstreaks

### LINKED ISSUE #30395: HIGHLANDER: Attacking a brain repeadly heals you and counts towards your kill streak.
Round ID 75662
I nearly won a round of highlander by discovering and exploiting this bug. Please don't ban me lmao.

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixes #30395 

Happened because dust() returns for brains.

### CODE REVIEW

File: code/game/objects/items/weaponry.dm [hunk 1]
```diff
@@ -117,7 +117,7 @@
 
 /obj/item/claymore/highlander/attack(mob/living/target, mob/living/user)
 	. = ..()
-	if(target && target.stat == DEAD && target.mind && target.mind.special_role == "highlander")
+	if(target && target.stat == DEAD && target.mind && target.mind.special_role == "highlander" && !isbrain(target))
```
Comment (ChangelingRain) [inline code/game/objects/items/weaponry.dm]: I think it'd be better for dust() to return TRUE/FALSE based on result, and check that.

Comment (Cyberboss) [inline code/game/objects/items/weaponry.dm]: First check should be !QDELETED(target)

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by WJohn** [comment]
Are there any other cases where this might happen? MMIs, intelicards, pAIs etc?

File: code/game/objects/items/weaponry.dm [hunk 2]
```diff
@@ -115,9 +115,9 @@
 	if(nuke_disk)
 		to_chat(user, "<span class='boldwarning'>It's holding the nuke disk!</span>")
 
-/obj/item/claymore/highlander/attack(mob/living/target, mob/living/user)
+/obj/item/claymore/highlander/attack(mob/living/carbon/target, mob/living/user)
```
Comment (ChangelingRain) [inline code/game/objects/items/weaponry.dm]: this doesn't do anything though

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**